internals/webhooks/validate/migrationhierarchy/migrationhierarchy_create.go

```diff
@@ -0,0 +1,249 @@
+package webhooks
+
+import (
+	"context"
+	"fmt"
+	danav1 "github.com/dana-team/hns/api/v1"
+	"github.com/dana-team/hns/internals/utils"
+	quotav1 "github.com/openshift/api/quota/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func (a *MigrationHierarchyAnnotator) handleCreate(mhObject *utils.ObjectContext, reqUser string) admission.Response {
+	ctx := mhObject.Ctx
+	logger := log.FromContext(ctx)
+
+	currentNSName := mhObject.Object.(*danav1.MigrationHierarchy).Spec.CurrentNamespace
+	currentNS, err := utils.NewObjectContext(ctx, a.Client, client.ObjectKey{Namespace: "", Name: currentNSName}, &corev1.Namespace{})
+	if err != nil {
+		logger.Error(err, "failed to create object", "currentNS", currentNSName)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	toNSName := mhObject.Object.(*danav1.MigrationHierarchy).Spec.ToNamespace
+	toNS, err := utils.NewObjectContext(ctx, a.Client, client.ObjectKey{Namespace: "", Name: toNSName}, &corev1.Namespace{})
+	if err != nil {
+		logger.Error(err, "failed to create object", "toNS", toNSName)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if rsp := utils.ValidateNamespaceExist(currentNS); !rsp.Allowed {
+		return rsp
+	}
+
+	if rsp := utils.ValidateNamespaceExist(toNS); !rsp.Allowed {
+		return rsp
+	}
+
+	currentNSArray := utils.GetNSDisplayNameArray(currentNS)
```

Comment:
change that to currentNamespaces type like array should not describe a variable

Reply:
changed